cszczepaniak/cribbly-backend
# Deploy to ECS

# What I added (link any issues addressed)
Phew. This was a long journey. Here's what I did:
1. I dockerized the application, including adding a docker-compose file for local development
    - The application can now be run locally with `make start`, which calls `docker-compose` under the hood
    - You will need to create `config/dev.env` before doing this. You can see what variables need to be set in `config/example.env` -- the password will be set in MySQL the first time you run `make start` so it can be anything
    - You will be able to access your local DB on localhost port 32000 with the password you set in `dev.env`
 2. I deployed the app to Amazon ECS with RDS following [this YouTube playlist](https://www.youtube.com/playlist?list=PL0dOL8Z7pG3IWsvseNd-JoFTHL16P_iTC)
    - Currently it's running at https://cribbly.oh-heck.com/api (click [here](https://cribbly.oh-heck.com/sample) to test out an endpoint) but this would be easy to change if we wanted to point a different domain (cribbly.net) at our load balancer
    - I did make our RDS instance public-facing (probably not entirely recommended); this will let us to connect to it from the outside world (I can make you an IAM user to be able to see stuff in AWS). I figured this would be handy for emergencies and debugging
    - This will deploy on each push to `main`, or we can run it manually

Maybe this isn't where we want to end up eventually. But I wanted to get _something_ out there that we can play with.

# How to test it
Test out the local dockerized version and make sure you can connect to the DB.

Try doing stuff with the deployed API. Register some players with /api/player/login, make some teams, etc. etc. See the data in our RDS instance (I can get you the password and whatever else to get connected).
